source/opt/propagator.h

```diff
@@ -198,6 +198,20 @@ class SSAPropagator {
   // Instruction::GetSingleWordOperand.
   bool IsPhiArgExecutable(ir::Instruction* phi, uint32_t i) const;
 
+  // Returns true if |inst| has a recorded status. This will be true once |inst|
+  // has been simulated once.
+  bool HasStatus(ir::Instruction* inst) const { return statuses_.count(inst); }
+
+  // Returns the current propagation status of |inst|.
+  PropStatus Status(ir::Instruction* inst) const {
+    return statuses_.find(inst)->second;
+  }
+
+  // Records the propagation status |status| for |inst|.
+  void SetStatus(ir::Instruction* inst, PropStatus status) {
+    statuses_[inst] = status;
```

Comment:
Move the checking for old status here, please. Better contained.

Reply:
Good idea, will do.